File: mycollab-services/src/main/java/com/mycollab/common/service/MonitorItemService.java
package com.mycollab.common.service;

import com.mycollab.cache.IgnoreCacheClass;
import com.mycollab.common.domain.MonitorItem;
import com.mycollab.common.domain.criteria.MonitorSearchCriteria;
import com.mycollab.db.persistence.service.ICrudService;
import com.mycollab.db.persistence.service.ISearchableService;
import com.mycollab.module.user.domain.SimpleUser;

import java.util.Collection;
import java.util.List;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
@IgnoreCacheClass
public interface MonitorItemService extends ICrudService<Integer, MonitorItem>, ISearchableService<MonitorSearchCriteria> {

    boolean isUserWatchingItem(String username, String type, Integer typeId);

    List<SimpleUser> getWatchers(String type, Integer typeId);

    void saveMonitorItems(Collection<MonitorItem> monitorItems);
}

File: mycollab-services/src/main/java/com/mycollab/common/service/MonitorItemService.kt
package com.mycollab.common.service

import com.mycollab.cache.IgnoreCacheClass
import com.mycollab.common.domain.MonitorItem
import com.mycollab.common.domain.criteria.MonitorSearchCriteria
import com.mycollab.db.persistence.service.ICrudService
import com.mycollab.db.persistence.service.ISearchableService
import com.mycollab.module.user.domain.SimpleUser

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
@IgnoreCacheClass
interface MonitorItemService : ICrudService<Int, MonitorItem>, ISearchableService<MonitorSearchCriteria> {

    fun isUserWatchingItem(username: String, type: String, typeId: Int?): Boolean

    fun getWatchers(type: String, typeId: Int?): List<SimpleUser>

    fun saveMonitorItems(monitorItems: Collection<MonitorItem>)
}
